packages/react-vapor/src/components/validation/hoc/WithNonEmptyValueMultiSelectValidationHOC.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {createStructuredSelector} from 'reselect';
+import {IDispatch} from '../../../utils/ReduxUtils';
+import {IMultiSelectOwnProps} from '../../select/MultiSelectConnected';
+import {SelectSelector} from '../../select/SelectSelector';
+import {ValidationActions} from '../ValidationActions';
+import {ValidationTypes} from '../ValidationTypes';
+import {InferableComponentEnhancer} from './connectHOC';
+
+const mapStateToProps = createStructuredSelector({
+    selectedValues: SelectSelector.getMultiSelectSelectedValues,
+});
+
+const mapDispatchToProps = (dispatch: IDispatch) => ({
+    setError: (id: string, error: string) => dispatch(ValidationActions.setError(id, error, ValidationTypes.nonEmpty)),
+    clearError: (id: string) => dispatch(ValidationActions.clearError(id, ValidationTypes.nonEmpty)),
+});
+
+export interface IWithNonEmptyValueMultiSelectValidationProps {
+    validationMessage?: string;
+    resetErrorOnUnmount?: boolean;
+    isException?: boolean;
+}
+
+export const withNonEmptyMultiSelectHOC = <T extends IMultiSelectOwnProps>(Component: React.ComponentType<T>) => {
+    type NewOwnProps = T & IWithNonEmptyValueMultiSelectValidationProps;
+    type StateProps = ReturnType<typeof mapStateToProps>;
+    type DispatchProps = ReturnType<typeof mapDispatchToProps>;
+    const WrappedMultiSelect: React.FunctionComponent<NewOwnProps & StateProps & DispatchProps> = ({
+        selectedValues,
+        setError,
+        clearError,
+        validationMessage = 'Selection is empty and should not be empty',
+        resetErrorOnUnmount,
+        isException,
+        ...props
+    }) => {
+        const hasValuesSelected = selectedValues.length > 0;
+
+        React.useEffect(() => {
+            clearError(props.id);
+            return () => {
+                resetErrorOnUnmount && clearError(props.id);
+            };
+        }, []);
+
+        React.useEffect(() => {
+            if (!isException) {
+                setError(props.id, !hasValuesSelected ? validationMessage : '');
+            }
+        }, [hasValuesSelected]);
```

Comment:
Not sure about the usefulness of `isException` (see previous comments), but if you keep it, it should also be included in the dependencies here

```suggestion
        }, [isException, hasValuesSelected]);
```